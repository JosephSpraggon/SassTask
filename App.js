import React, { useState } from 'react';
import { Keyboard, Image, KeyboardAvoidingView, Platform, StyleSheet, Text, TextInput, TouchableOpacity, View } from 'react-native';
import TaskList from './screens/TaskList';
import {Container,Content} from 'native-base';
import Swiper from 'react-native-swiper';
import Monster from './screens/Monster';
import TaskPage from './screens/TaskPage';

export default function App() {
  const [completedTasks, setCompletedTasks] = useState([]);
  const [incompleteTasks, setIncompletedTasks] = useState([]);
  const [message, setMessage] = useState("");
  const [imageSrc, setImageSrc] = useState(require('./assets/Sassagotchi-egg-coloured.gif'));

  const monsterPicker = () => {
    if(completedTasks.length >= 3) {
      setImageSrc('./assets/SnapSquash.gif')
    } else if(completedTasks >= 1) {
      setImageSrc('./assets/SasSquash.gif')
    } else {
      setImageSrc('./assets/Sassagotchi-egg-coloured.gif')
    }
  }

  return (
    <Container style={styles.EntireApp}>
      <Swiper
      loop ={false}
      showsPagination={false}
      index={1}
      onIndexChanged={monsterPicker()}>
        <Monster />
        <TaskList setCompletedTasks={setCompletedTasks}
                  setIncompletedTasks={setIncompletedTasks}
                  completedTasks={completedTasks}
                  incompleteTasks={incompleteTasks}
                  message={message}
                  setMessage={setMessage} />
        <TaskPage completedTasks={completedTasks}
         incompleteTasks={incompleteTasks}
          message={message}
          imageSrc={imageSrc}
           />
      </Swiper>
    </Container>
  );
}

const styles = StyleSheet.create({
  EntireApp:{
    flex: 1,
    backgroundColor: 'lightsalmon'
  }
});
