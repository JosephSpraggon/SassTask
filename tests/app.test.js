import React from 'react';
import renderer from 'react-test-renderer';

import App from '../App';

test('it works', () => {
  expect(true).toBeTruthy();
});

describe('<App />', () => {
  const tree = renderer.create(<App />).toJSON();

  it('has 1 child', () => {
    expect(tree.children.length).toBe(3);
  });

  it('renders correctly', () => {
    expect(tree).toMatchSnapshot();
  });

});
