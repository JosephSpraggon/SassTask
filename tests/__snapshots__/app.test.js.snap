// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "lightsalmon",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "backgroundColor": "blue",
        }
      }
    >
      Team SassTask!
    </Text>
  </View>
  <View>
    <View>
      <Text>
        task1
      </Text>
    </View>
    <View>
      <Text>
        task2
      </Text>
    </View>
    <View>
      <Text>
        dj khalid
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
        "margin": 10,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 60,
          "margin": 5,
          "padding": 20,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        placeholder="add task"
        rejectResponderTermination={true}
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#f05365",
          "borderColor": "grey",
          "borderRadius": 60,
          "display": "flex",
          "fontSize": 40,
          "height": 40,
          "margin": 10,
          "opacity": 1,
          "textAlign": "center",
          "width": 40,
        }
      }
    >
      +
    </View>
  </View>
</View>
`;
